ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java

```diff
@@ -247,11 +248,23 @@ public void startup() throws InstantiationException {
           Utils.getObj(serverConfig.serverStoreKeyConverterFactory, properties, registry);
 
       Predicate<MessageInfo> skipPredicate = new ReplicationSkipPredicate(accountService, replicationConfig);
-      replicationManager =
-          new ReplicationManager(replicationConfig, clusterMapConfig, storeConfig, storageManager, storeKeyFactory,
-              clusterMap, scheduler, nodeId, connectionPool, registry, notificationSystem, storeKeyConverterFactory,
-              serverConfig.serverMessageTransformer, clusterParticipants.get(0), skipPredicate);
-      replicationManager.start();
+      switch(replicationConfig.replicationManager) {
+        case ReplicationConfig.DEFAULT_REPLICATION_MANAGER:
```

Comment:
Since you are using the classpath of the replication manager class, you can use reflection (look at Utils.getObj).